src/checks/unordered_key.rs

```diff
@@ -203,4 +209,18 @@ mod tests {
 
         run_unordered_tests(asserts);
     }
+
+    #[test]
+    fn two_ordered_groups_with_two_substitution_keys_test() {
```

Comment:
Could you add a few tests with multiple substitution keys in one variable and also with the wrong order of the keys (for example, between or before substitution keys) ?

Reply:
Sure, I'll do it. Thanks for feedback.

Reply:
Done.

Reply:
What about this case:
> CCC=$HHH
> BBB=XYZ
> HHH=VAL
> ZZZ=VALUE

Should warning tell us that CCC should be placed below HHH or leave it as is with "The BBB key should go before the CCC key"?

Reply:
And what right order should be for this case:

> ZZZ=$AAA
> BBB=III
> AAA=VVV

1)
> AAA=VVV
> BBB=III
> ZZZ=$AAA

or

2)
> AAA=VVV
> ZZZ=$AAA
> BBB=III

Reply:
Good questions!
I think for such cases it's not worth changing the order, because it may depend on an external variable.

I.e for this one:
```sh
CCC=$HHH
BBB=XYZ
HHH=VAL
ZZZ=VALUE
```
the correct version should look like this:
```sh
BBB=XYZ
CCC=$HHH # Uses external HHH key
HHH=VAL
ZZZ=VALUE
```

But it might be better to consider this in a separate issue 🤔

Reply:
> the correct version should look like this:
> 
> ```shell
> BBB=XYZ
> CCC=$HHH # Uses external HHH key
> HHH=VAL
> ZZZ=VALUE
> ```

I don't fully understand why you think `$HHH` external? The next line is `HHH=VAL`.
Or I missed something?

Reply:
> I don't fully understand why you think `$HHH` external? The next line is `HHH=VAL`.
> Or I missed something?

Because original `.env` has next order (in example above):
```shell
CCC=$HHH  # HHH can be external or not (we don't know for sure)
BBB=XYZ
HHH=VAL # this can override HHH key 
ZZZ=VALUE
```

And it is likely that the key depends on an external variable (or from another sort group).
But if we create `Warning` - this can cause confusion and for example  and a fixer will fix it.
Thus breaking the relationship between `CCC` and `HHH`

At the same time, these are complex validation rules, and in such cases, I think it would be more correct for the user to divide into groups (if necessary):

```shell
CCC=$HHH

BBB=XYZ
HHH=VAL
ZZZ=VALUE
```

And in this case, the current implementation already suits us completely.
This is controversial, so I suggested discussing it in another issue if needed.